### PR TITLE
Add examples for best practices for section 4.2.4 microflow complexity

### PR DESCRIPTION
**Pull request by Yiyun333**
This is based on GitHub issue #5152

### CODE REVIEW

**Review by reinouts — COMMENTED** [review]
- The caption of the decision should probably reflect that it checks both if `$Parameter`  is `true` and whether `$DeviceType` equals `Phone`.
- To make the difference between the two examples clearer, the picture should probably show both microflows in which the decisions appear and the expressions inside the decisions.

**Comment by Yiyun333** [comment]
> * The caption of the decision should probably reflect that it checks both if `$Parameter`  is `true` and whether `$DeviceType` equals `Phone`.
> * To make the difference between the two examples clearer, the picture should probably show both microflows in which the decisions appear and the expressions inside the decisions.
@reinouts Thank you very much for the comments. I just updated the doc accordingly. Please let me know if there are more things to improve.

**Comment by reinouts** [comment]
Still has to be reviewed by the doc team but from a technical/theory point of view I think this is fine.

**Comment by Yiyun333** [comment]
> Still has to be reviewed by the doc team but from a technical/theory point of view I think this is fine.

@reinouts Great! Thanks a lot!

File: content/en/docs/howto/general/dev-best-practices.md [hunk 1]
```diff
@@ -335,7 +335,11 @@ If you decide to color code the different activities in your app, be sure to ali
...
 
-Nested `IF` statements in a single microflow expression are not recommended. If multiple checks depend on one another, this should be represented by multiple decisions in the microflow, so that the complexity is not hidden away in the expressions. You can use `AND` and `OR` operators to produce complex expressions if necessary.
+Nested `IF` statements in a single microflow expression are not recommended. If multiple checks depend on one another, this should be represented by multiple decisions in the microflow, so that the complexity is not hidden away in the expressions. You can use `AND` and `OR` operators to produce complex expressions if necessary. Examples of what is not recommended and what is recommended are shown below. If necessary, the single microflow expression in the following example can also be written as `if ($currentDeviceType = System.DeviceType.Phone and $Parameter = true) then true else false` using the `AND` operator. However, we still recommend the low-code approach shown below, which makes it easier to understand what is really happening.
+
+{{< figure src="/attachments/howto/general/dev-best-practices/not-recommended-microflow.png" width="450px">}}
```
Comment (MariaShaposhnikova) [inline content/en/docs/howto/general/dev-best-practices.md]: It's not the best practice to place two images one under the other. I would make it explicit in the text. I would also suggest to swap the order which they are presented and start from the recommended approach. 
For example:
In the example below, there is a low-code approach which makes it easier to understand what is happening in the microflow:
Image
The example below shows a possible but not recommended approach: 
Image
I would also remove exclamation marks, a cross mark, and a check mark from the images to tone it down a bit + these icons are not standardized.

File: content/en/docs/howto/general/dev-best-practices.md [hunk 2]
```diff
@@ -335,7 +335,11 @@ If you decide to color code the different activities in your app, be sure to ali
 
 #### 4.2.4 Complexity
 
-Nested `IF` statements in a single microflow expression are not recommended. If multiple checks depend on one another, this should be represented by multiple decisions in the microflow, so that the complexity is not hidden away in the expressions. You can use `AND` and `OR` operators to produce complex expressions if necessary.
+Nested `IF` statements in a single microflow expression are not recommended. If multiple checks depend on one another, this should be represented by multiple decisions in the microflow, so that the complexity is not hidden away in the expressions. You can use `AND` and `OR` operators to produce complex expressions if necessary. Examples of what is not recommended and what is recommended are shown below. If necessary, the single microflow expression in the following example can also be written as `if ($currentDeviceType = System.DeviceType.Phone and $Parameter = true) then true else false` using the `AND` operator. However, we still recommend the low-code approach shown below, which makes it easier to understand what is really happening.
```
Comment (MariaShaposhnikova) [inline content/en/docs/howto/general/dev-best-practices.md]: I would start describing the examples with a new paragraph.

Comment (Yiyun333) [reply content/en/docs/howto/general/dev-best-practices.md]: @MariaShaposhnikova Many thanks for the review and advice. I just updated the doc based on your suggestions. Please let me know if you have any further advice on the updated version.

File: content/en/docs/howto/general/dev-best-practices.md [hunk 3]
```diff
@@ -337,11 +337,11 @@ If you decide to color code the different activities in your app, be sure to ali
...
 
-The example below shows a low-code approach we recommend because it presents a clear picture of what is happening in the microflow:
+The example below shows a low-code approach that we recommend because it presents a clear picture of what is happening in the microflow:
 
 {{< figure src="/attachments/howto/general/dev-best-practices/recommended-microflow.png" width="700px">}}
 
-The example below shows an approach we do not recommend. We can rewrite the microflow expression in this example as `if ($currentDeviceType = System.DeviceType.Phone and $Parameter = true) then true else false` using the `AND` operator. However, it is still not clear enough and the low-code approach shown in the above example is better.
+The example below shows an approach that we do not recommend. We can rewrite the microflow expression in this example as `if ($currentDeviceType = System.DeviceType.Phone and $Parameter = true) then true else false` using the `AND` operator. However, it is still not clear enough and the low-code approach shown in the above example is better.
```
Comment (MariaShaposhnikova) [inline content/en/docs/howto/general/dev-best-practices.md]:
```suggestion
The example below shows an approach that we do not recommend. You can rewrite the microflow expression in this example as `if ($currentDeviceType = System.DeviceType.Phone and $Parameter = true) then true else false` using the `AND` operator. However, it is still not clear enough and the low-code approach shown in the above example is preferable.
```

**Comment by MariaShaposhnikova** [comment]
@Yiyun333 , thanks for implementing all comments! I added one suggestion, the rest looks good!

**Comment by Yiyun333** [comment]
> @Yiyun333 , thanks for implementing all comments! I added one suggestion, the rest looks good!

@MariaShaposhnikova Thanks for the review!

**Pull request merged**